plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -254,7 +253,7 @@ public class CpsFlowExecution extends FlowExecution implements BlockableResume {
      * Any additional scripts {@linkplain CpsGroovyShell#parse(GroovyCodeSource) parsed} afterward, keyed by
      * their FQCN.
      */
-    /*package*/ /*final*/ Map<String,String> loadedScripts = new HashMap<>();
+    /*package*/ /*final*/ Map<String,String> loadedScripts = new LinkedHashMap<>();
```

Comment:
Maybe even

```suggestion
    /*package*/ /*final*/ LinkedHashMap<String,String> loadedScripts = new LinkedHashMap<>();
```

since it is intended to retain insertion order.

Reply:
Unfortunately no, because previously serialized `build.xml`s would include a `HashMap`.

Reply:
My first thought would be a `TreeMap` but I guess this works just as well.

Reply:
My first thought _was_ a `TreeMap`. However it would need to use a special `Comparator` since then for example `Script10` would come before `Script2`. `LinkedHashMap` seems simpler and safer; if the original `evaluate`s in the prior Jenkins session loaded scripts successfully in a certain order, after resumption we should try to load them in the same order.